SubmitHGCalPGun.py

```diff
@@ -395,26 +416,69 @@ def submitHGCalProduction(*args, **kwargs):
             s_template=s_template.replace('DUMMYINPUTFILELIST',inputFiles)
             s_template=s_template.replace('DUMMYEVTSPERJOB',str(-1))
 
-
-
         if (opt.DTIER == 'NTUP'):
             s_template=s_template.replace('DUMMYRECLUST',str(opt.RECLUST))
             s_template=s_template.replace('DUMMYSGO',str(opt.ADDGENORIG))
             s_template=s_template.replace('DUMMYSGE',str(opt.ADDGENEXTR))
             s_template=s_template.replace('DUMMYSPFC',str(opt.storePFCandidates))
             s_template=s_template.replace('DUMMYMULCLUSTAG', str(opt.MULTICLUSTAG))
 
+        if (opt.DTIER == 'ALL'):
+
+            sr_template=r_template
+            sn_template=n_template
+
+            sr_template=sr_template.replace('DUMMYINPUTFILELIST',"'file:"+outfile+"'")
+            outfile = outfile.replace('GSD','RECO')
+
+            sr_template=sr_template.replace('DUMMYFILENAME',outfile)
+            sr_template=sr_template.replace('DUMMYDQMFILENAME',outdqmfile)
+            sr_template=sr_template.replace('DUMMYSEED',str(job))
+            sr_template=sr_template.replace('DUMMYINPUTFILELIST',outfile)
+            sr_template=sr_template.replace('DUMMYEVTSPERJOB',str(-1))
+
+            sn_template=sn_template.replace('DUMMYINPUTFILELIST',"'file:"+outfile+"'")
+            outfile = outfile.replace('RECO','NTUP')
+            sn_template=sn_template.replace('DUMMYFILENAME',outfile)
+            sn_template=sn_template.replace('DUMMYDQMFILENAME',outdqmfile)
+            sn_template=sn_template.replace('DUMMYSEED',str(job))
+            sn_template=sn_template.replace('DUMMYEVTSPERJOB',str(-1))
+
+            sn_template=sn_template.replace('DUMMYRECLUST',str(opt.RECLUST))
+            sn_template=sn_template.replace('DUMMYSGO',str(opt.ADDGENORIG))
+            sn_template=sn_template.replace('DUMMYSGE',str(opt.ADDGENEXTR))
+            sn_template=sn_template.replace('DUMMYSPFC',str(opt.storePFCandidates))
+            sn_template=sn_template.replace('DUMMYMULCLUSTAG', str(opt.MULTICLUSTAG))
+            
+
+
         # submit job
         # now write the file from the s_template
+        
         cfgfile_path = outDir + '/cfg/' + cfgfile
         write_template= open(cfgfile_path, 'w')
         write_template.write(s_template)
-        write_template.close()
```

Comment:
You delete the line closing the file, but never add it anywhere else?

Reply:
Yes, sorry this is clearly a mistake, should be fixed now.